crates/sui-core/src/checkpoints/mod.rs

```diff
@@ -1290,33 +1291,8 @@ impl CheckpointSignatureAggregator {
         &mut self,
         data: CheckpointSignatureMessage,
     ) -> Result<AuthorityStrongQuorumSignInfo, ()> {
-        let their_digest = *data.summary.digest();
         let (_, signature) = data.summary.into_data_and_sig();
         let author = signature.authority;
-        // consensus ensures that authority == narwhal_cert.author
-        if their_digest != self.digest {
-            if let InsertResult::QuorumReached(data) =
-                self.failures.insert_generic(author, signature)
-            {
-                panic!(
-                    "Checkpoint fork detected - f+1 validators submitted checkpoint digest at seq {} different from our digest {}. \
-                     Our checkpoint summary: {:?}. Validators with different digests: {:?}",
-                    self.summary.sequence_number,
-                    self.digest,
-                    self.summary,
-                    data.keys()
-                );
-            }
-            warn!(
-                checkpoint_seq = self.summary.sequence_number,
-                "Validator {:?} has mismatching checkpoint digest {}, we have digest {}",
-                author.concise(),
-                their_digest,
-                self.digest
-            );
```

Comment:
You don't think we should keep this warning? My understanding is after this merges:
1. We won't know of any discrepancy until it is a confirmed fork. 
2. We will be reliant on external validators telling us that they forked themselves. If we keep the above logs, it will at least show up in mysten validator logs that some other validator is proposing bad checkpoints/may have forked. Considering that one potential mitigation to a fork is to resync, this could be our only method of detection in some cases.

Either that or we could bump a metric, but given that this shouldn't happen on honest validators, logging feels more appropriate to me

Reply:
we could add a prom counter here and then we'd know right away. it would always, ideally, be zero, but would increment monotonically.